woocommerce/woocommerce-ios
# [Mobile Payments] Incorrect amount when collecting Simple Payment with RTL currency

Closes: #10472 
P2: peaMlT-bw-p2
Ref: p1695726363759459-slack-C025A8VV728

## Description
This PR removes the currency symbol from the amount we send to the Order stores when updating a Simple Payments order. 

## Why
This is done so there's no encoding/decoding issues in certain cases, for example currencies using special characters, with a `.` in their symbol, and/or using `right` currency position were causing that we send the wrong amount to the order remote when updating the simple payments order.

## Testing instructions
1. In wp-admin > WooCommerce > Settings > Currency options, change the the following settings: and save the changes:
- Currency: AED
- Currency position: Right

<img width="970" alt="Screenshot 2023-09-26 at 16 41 54" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/7fe2a4c6-b8da-4620-b375-b07daab54ed7">

2. In the app, go to Menu > Payments > Collect Payment > Enter an AED amount, for example `1`, or `12.20` since also accepts two decimals, tap "Next"
3. See in proxyman the following request and `total` fitting the amount you entered :
```
path=/wc/v3/orders&_method=post

     {
        "id": 62,
        "name": "Simple Payments",
        "tax_class": "",
        "tax_status": "taxable",
        "amount": "",
        "total": "1.00",
        "total_tax": "0.00",
        "taxes": [],
        "meta_data": []
      }
```
 4. Tap `Take Payment`
 5. See in proxyman the following request and `total` still fitting the amount you entered :
```
path=/wc/v3/orders/83&_method=post

  {
        "id": 62,
        "name": "Simple Payments",
        "tax_class": "",
        "tax_status": "taxable",
        "amount": "",
        "total": "1.00",
        "total_tax": "0.00",
        "taxes": [],
        "meta_data": []
      }
```
6. Go back to Orders, confirm the correct order and amount have been created.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/3812076/e170e756-f9d8-4e91-ba04-089155341692
